hexrd/material.py

```diff
@@ -122,88 +184,297 @@ def __str__(self):
+        self._atominfo[:,4] = 8.0 * (numpy.pi**2) * (self._atominfo[:,4]**2) * 1E-2
+
+        '''
+        get atome types here i.e. the atomic number of atoms at each site
+        '''
+        atype = '_atom_site_type_symbol'
+        patype = (atype in cifdata)
+        if(not patype):
+            raise RuntimeError('atom types not defined in cif file.')
+
+        satype = cifdata[atype]
+        atomtype = []
+
+        for s in satype:
+            atomtype.append(ptable[s])
+
+        self._atomtype  = numpy.asarray(atomtype).astype(numpy.int32)
+        self._sgsetting = 0
+
+    def _readHDFxtal(self, fhdf=DFLT_NAME, xtal=DFLT_NAME):
+        """
+        >> @AUTHOR:     Saransh Singh, Lawrence Livermore National Lab, saransh1@llnl.gov
+        >> @DATE:       10/17/2019 SS 1.0 original
+        >> @DETAILS:    hexrd3 will have real structure factors and will require the overhaul
+                        of the crystallography. In this effort, we will have a HDF file reader.
+                        the file will be the same as the EMsoft xtal file. h5py will be used for
+                        i/o
+        """
+
+        fexist = path.exists(fhdf)
+        if(fexist):
+            fid = h5py.File(fhdf, 'r')
+            xtal = "/"+xtal
+            if xtal not in fid:
+                raise IOError('crystal doesn''t exist in material file.')
+        else:
+            raise IOError('material file does not exist.')
+
+        gid         = fid.get(xtal)
+
+        sgnum       = numpy.asscalar(numpy.array(gid.get('SpaceGroupNumber'), \
+                                    dtype = numpy.int32))
+        """
+            IMPORTANT NOTE:
+            note that the latice parameters is nm by default
+            hexrd on the other hand uses A as the default units, so we
+            need to be careful and convert it right here, so there is no
+            confusion later on
+        """
+        lparms      = list(gid.get('LatticeParameters'))
+
+        for i in range(6):
+            if(i < 3):
+                lparms[i] = _angstroms(lparms[i]*10.0)
+            else:
+                lparms[i] = _degrees(lparms[i])
+
+        self._lparms    = lparms
+        #self._lparms    = self._toSixLP(sgnum, lparms)
+        # fill space group and lattice parameters
+        self.sgnum      = sgnum
+
+        # the U factors are related to B by the relation B = 8pi^2 U
+        self._atominfo  = numpy.transpose(numpy.array(gid.get('AtomData'), dtype = numpy.float64))
+        self._U         = numpy.transpose(numpy.array(gid.get('U'), dtype = numpy.float64))
+
+        # read atom types (by atomic number, Z)
+        self._atomtype = numpy.array(gid.get('Atomtypes'), dtype = numpy.int32)
+        self._atom_ntype = self._atomtype.shape[0]
+
+        self._sgsetting = numpy.asscalar(numpy.array(gid.get('SpaceGroupSetting'), \
+                                        dtype = numpy.int32))
+        self._sgsetting -= 1
```

Comment:
@saransh13 It looks like if I load in the HDF5 file and then dump it out immediately, the space group setting drops in value by 1, maybe because of this line here.

CeO2 for instance starts with "1". After load and dump, it is then "0". After load and dump again, it is then "-1".

This is the only main difference I found for round-tripping.

Reply:
`sgsetting` should always be 0. All calculations are performed in the standard setting, corresponding to the value 0. 
Will push a fix shortly.

Reply:
The problem was that that `sgsetting` was stored as "1" based indexing in the h5 file but used in "0" based indexing in the class. Everything is "0" based now. I've pushed the changes and also uploaded a "material_new.h5" on the hexrd google drive where everything is "0" based. You should use that file for testing instead.

Reply:
@saransh13 Sounds great. I'll try it out. Thank you Saransh!